### PR TITLE
Add set to current context name

### PR DESCRIPTION
**Pull request by bzd111**
Add set to current context name

### CODE REVIEW

File: cmd/kconf.go [hunk 1]
```diff
@@ -141,6 +141,49 @@ var viewCmd = &cobra.Command{
...
+		hasContext := false
+		for _, ctx := range contexts {
+			if contextName == ctx {
+				hasContext = true
+			}
+		}
+		if hasContext {
+			err := config.WriteCurrentContext(contextName)
+			if err == nil {
+				fmt.Printf("switch %s success", contextName)
```
Comment (particledecay) [inline cmd/kconf.go:177]: I would prefer `"Using context '%s'"`, keeping in line with both the capitalization elsewhere and the verbiage of the "use" command.

File: cmd/kconf.go [hunk 2]
```diff
@@ -141,6 +141,49 @@ var viewCmd = &cobra.Command{
...
+		contextName := args[0]
+
+		config, err := kubeconfig.GetConfig()
+		if err != nil {
+			log.Fatal().Msgf("Could not read main config")
+		}
+
+		contexts, currentContext := config.List()
+		if currentContext == contextName {
+			fmt.Printf("current context is %s, don't need switch", currentContext)
```
Comment (particledecay) [inline cmd/kconf.go:164]: `"Current context is already '%s'"`

File: cmd/kconf.go [hunk 3]
```diff
@@ -141,6 +141,49 @@ var viewCmd = &cobra.Command{
...
+		}
+		if hasContext {
+			err := config.WriteCurrentContext(contextName)
+			if err == nil {
+				fmt.Printf("switch %s success", contextName)
+			} else {
+				log.Fatal().Msgf("Could not save current context to main config")
+			}
+		} else {
+			fmt.Println("There isn't such as this context name:", contextName)
```
Comment (particledecay) [inline cmd/kconf.go:182]: `"Context '%s' not found"`

File: cmd/kconf.go [hunk 4]
```diff
@@ -141,6 +141,49 @@ var viewCmd = &cobra.Command{
 	},
 }
 
+var useCmd = &cobra.Command{
+	Use:   "use",
+	Short: "set current context",
+	Long:  `set current context by user provide the name`,
```
Comment (particledecay) [inline cmd/kconf.go:147]: `set the current context in the main kubeconfig`

File: cmd/kconf.go (hunk 1 above)
Comment (bzd111) [reply cmd/kconf.go:177]: ok, I will change it.

File: cmd/kconf.go (hunk 2 above)
Comment (bzd111) [reply cmd/kconf.go:164]: Ok, I will change it.

File: cmd/kconf.go (hunk 3 above)
Comment (bzd111) [reply cmd/kconf.go:182]: Ok, I will change it.

File: cmd/kconf.go (hunk 4 above)
Comment (bzd111) [reply cmd/kconf.go:147]: OK, I will change it.

File: cmd/kconf.go [hunk 5]
```diff
@@ -141,6 +141,49 @@ var viewCmd = &cobra.Command{
...
+		contexts, currentContext := config.List()
+		if currentContext == contextName {
+			fmt.Printf("current context is %s, don't need switch", currentContext)
+			return
+		}
+
+		hasContext := false
+		for _, ctx := range contexts {
+			if contextName == ctx {
+				hasContext = true
```
[the remark this answers is not in the document]
Comment (bzd111) [reply cmd/kconf.go:171]: I will add `break` statement

**Comment by bzd111** [comment]
I have changed by your request.Please review it.

**Comment by particledecay** [comment]
@bzd111 everything looks good, would you please mind updating your branch?

Btw, it's usually better to submit PRs from a feature branch in your forked repo. It's not required, but just noticing you're submitting from your repo's master branch. Please don't rebase, do a fetch and merge from upstream/master.

**Comment by bzd111** [comment]
OK,I will checkout branch from upstream/master,then pick up commit to this branch ,then submit a PR

**Comment by bzd111** [comment]
I create a new [PR](https://github.com/particledecay/kconf/pull/11).Please close this PR.

**Comment by particledecay** [comment]
Superceded by #11

**Pull request closed without merging**